packages/block-library/src/navigation-link/edit.js

```diff
@@ -170,6 +171,14 @@ function NavigationLinkEdit( {
 								className="wp-block-navigation-link__inline-link-input"
 								value={ link }
 								showInitialSuggestions={ true }
+								showCreatePages={ true }
+								createEmptyPage={ ( pageTitle ) =>
+									savePage( {
+										title: pageTitle,
+										content: '',
+										status: 'publish', // TODO: use publish?
```

Comment:
Publish seems to be a good default for now.

Initially I thought about maybe inheriting the status from the current post/page and publish the new page together with the current post, but that seems fragile considering future edits.